awsagentprovider/src/test/java/software/amazon/opentelemetry/javaagent/providers/AwsTracerCustomizerProviderTest.java

```diff
@@ -0,0 +1,54 @@
+package software.amazon.opentelemetry.javaagent.providers;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.AfterEach;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class AwsTracerCustomizerProviderTest {
+
+    private final Map<String, String> originalProperties = new HashMap<>();
+
+    @BeforeEach
+    void setUp() throws ClassNotFoundException {
+        // Backup original system properties
+        backupProperty("otel.aws.application.signals.enabled");
+        backupProperty("otel.metrics.exporter");
+        backupProperty("otel.logs.export");
+        backupProperty("otel.aws.application.signals.exporter.endpoint");
+        backupProperty("otel.exporter.otlp.protocol");
+        backupProperty("otel.exporter.otlp.traces.endpoint");
+        backupProperty("otel.traces.sampler");
+        backupProperty("otel.traces.sampler.arg");
+    }
+
+    private void backupProperty(String key) {
+        if (System.getProperty(key) != null) {
+            originalProperties.put(key, System.getProperty(key));
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        // Restore original system properties
+        for (Map.Entry<String, String> entry : originalProperties.entrySet()) {
+            System.setProperty(entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Test
+    void testSetSystemPropertyFromEnvOrDefault_AllValuesSet() {
```

Comment:
I am not sure what this test does. It doesn't call `setSystemPropertyFromEnvOrDefault`?